slave/slave.c

```diff
@@ -30,58 +70,80 @@ query_result *get_vector(u_int vec_id)
 
     /* Open file in binary mode. */
     fp = fopen(filename, "r");
+
     if (fp == NULL) {
         exit_code = EXIT_FAILURE ^ vec_id;
     }
+
     else {
-        int num_elts = 4; // XXX: should be empirically determined average vector length
+        /* TODO: should be empirically determined average vector length */
+        int num_elts = 4;
         vector_val = (u_int64_t *) malloc(sizeof(u_int64_t) * num_elts);
         char buf[32];
+
         while (fgets(buf, 32, fp) != NULL) {
             if (vector_len > num_elts) {
                 num_elts *= 2;
                 vector_val = (u_int64_t *) realloc(vector_val, num_elts * sizeof(u_int64_t));
             }
+
             vector_val[vector_len++] = (u_int64_t) strtol(buf, NULL, 10);
         }
+
         fclose(fp);
     }
+
     query_result *vector = (query_result *) malloc(sizeof(query_result));
     vector->vector.vector_val = vector_val;
     vector->vector.vector_len = vector_len;
     vector->exit_code = exit_code;
+
     return vector;
 }
 
-query_result *rq_pipe_1_svc(rq_pipe_args query, struct svc_req *req)
+query_result *rq_pipe_1_svc(rq_pipe_args query, vector_clock recv_clock, struct svc_req *req)
 {
+    if (slave_clock == NULL) {
+        slave_clock = create_vclock();
+    }
+
+    handle_recvd_vclock(slave_clock, recv_clock, slave_id);
+
     query_result *this_result;
     query_result *next_result = NULL;
 
     u_int exit_code = EXIT_SUCCESS;
     this_result = get_vector(query.vec_id);
+
     /* Something went wrong with reading the vector. */
     if (this_result->exit_code != EXIT_SUCCESS) {
+        prep_send_vclock(slave_clock, slave_id);
+        this_result->vclock = slave_clock;
+
         return this_result;
     }
+
     /* We are the final call. */
     else if (query.next == NULL) {
+        prep_send_vclock(slave_clock, slave_id);
+        this_result->vclock = slave_clock;
+
         return this_result;
     }
 
     /* Recursive Query */
     else {
-
         char *host = query.machine_addr;
         CLIENT *client;
-        client = clnt_create(host,
-            REMOTE_QUERY_PIPE, REMOTE_QUERY_PIPE_V1, "tcp");
+        client = clnt_create(host, REMOTE_QUERY_PIPE, REMOTE_QUERY_PIPE_V1, "tcp");
```

Comment:
I don't think this is less than 80 characters?

Reply:
Fixed (along with a number of other lines in the same file.).